tests/unittests/CoreGraphics.drawing/CGContextDrawing_WidenPathTests.cpp

```diff
@@ -0,0 +1,359 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include "DrawingTest.h"
+
+CGLineCap lineCaps[] = { kCGLineCapButt, kCGLineCapSquare, kCGLineCapRound };
+CGLineJoin lineJoins[] = { kCGLineJoinMiter, kCGLineJoinBevel, kCGLineJoinRound };
+CGFloat miterLimits[] = { 0, 1, 5 };
+bool shouldUseContext[] = { false, true };
+bool addLineAfterWidening[] = { false, true };
+
+class CGPathCopyByStroking_Lines : public WhiteBackgroundTest<>,
+                                   public ::testing::WithParamInterface<::testing::tuple<CGLineCap, CGLineJoin, CGFloat, bool, bool>> {
+    CFStringRef CreateOutputFilename() {
+        CGLineCap lineCap = ::testing::get<0>(GetParam());
+        CGLineJoin lineJoin = ::testing::get<1>(GetParam());
+        CGFloat miterLimit = ::testing::get<2>(GetParam());
+        bool useContext = ::testing::get<3>(GetParam());
+        bool addLine = ::testing::get<4>(GetParam());
+
+        char* lineCapName;
+        switch (lineCap) {
+            case kCGLineCapButt:
+                lineCapName = "LineCapButt";
+                break;
+            case kCGLineCapSquare:
+                lineCapName = "LineCapSquare";
+                break;
+            case kCGLineCapRound:
+                lineCapName = "LineCapRound";
+                break;
+            default:
+                break;
+        }
+
+        char* lineJoinName;
+        switch (lineJoin) {
+            case kCGLineJoinMiter:
+                lineJoinName = "LineJoinMiter";
+                break;
+            case kCGLineJoinBevel:
+                lineJoinName = "LineJoinBevel";
+                break;
+            case kCGLineJoinRound:
+                lineJoinName = "LineJoinRound";
+                break;
+            default:
+                break;
+        }
+
+        return CFStringCreateWithFormat(nullptr,
+                                        nullptr,
+                                        CFSTR("TestImage.CopyByStroking_Lines.%f.%s.%s.%s.png"),
+                                        miterLimit,
+                                        lineJoinName,
+                                        lineCapName,
+                                        addLine ? "AddLine" : "NoLine",
```

Comment:
what happens when you add a line from a completed stroked path is sorta an implementation detail; you could consider dropping that parameter from every test to half the search space.